### PR TITLE
Remove EditorFeatures dependency to ImageCatalog

### PR DESCRIPTION
**Pull request by JieCarolHu**
fixed https://github.com/dotnet/roslyn/issues/26642
<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes
https://github.com/dotnet/roslyn/issues/26642

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

File: src/VisualStudio/CSharp/Test/GlyphExtensionsTests.cs [hunk 1]
```diff
@@ -169,6 +170,14 @@ public void TestWithEventsMemberGlyph()
                 isWithEvents: true);
         }
 
+        [Fact, WorkItem(26642, "https://github.com/dotnet/roslyn/issues/26642")]
+        public void TestNoReferenceToImageCatalog()
```
Comment (jasonmalinowski) [inline src/VisualStudio/CSharp/Test/GlyphExtensionsTests.cs]: We probably should have this in EditorServicesTests somewhere. If you need to make a new class, that's fine.

Comment (jasonmalinowski) [reply src/VisualStudio/CSharp/Test/GlyphExtensionsTests.cs]: We also probably should have a comment saying why this is important...

Comment (jasonmalinowski) [inline src/VisualStudio/CSharp/Test/GlyphExtensionsTests.cs]: I should also say: I encouraged @JieCarolHu to write this test no matter the method taken to achieve this fix;, that way nobody innocently adds the assembly reference back and breaks it again.

**Comment by KirillOsenkov** [comment]
Please also remove the reference from the .csproj file.

**Comment by jasonmalinowski** [comment]
@KirillOsenkov Is there a way to do that without inlining hundreds of constants? We went with this approach because it ensures what you want (and is verified by test) without hurting maintainability of the code.

**Comment by KirillOsenkov** [comment]
I think this is great as is.

Just not clear about the actual reference in the .csproj?

**Comment by KirillOsenkov** [comment]
Never mind. I just realized that you still need the reference at compile time, but it achieves the desired effect of erasing the assembly reference in the metadata, and the unit-test is there to verify it.

Perfect! Thanks.

**Comment by jinujoseph** [comment]
Approved to merge for 15.8.Preview4

**Pull request merged**